### PR TITLE
(PEAR-1764): properly use mantine Switch component built in label

### PR DESCRIPTION
**Pull request by ocshawn**
## Description

- change instances of mantine Switch component to use built in label in preparation for mantine patch 
- convert SwitchSpring to use mantine Switch component
- adjust tests and test ids to better match new SwitchSpring code

## Checklist

- [x] Added proper unit tests
- [x] Left proper TODO messages for any remaining tasks
- [x] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

**Comment by ocshawn** [comment]
the issues found in wave will be fixed when we update mantine

File: packages/portal-proto/src/components/SwitchSpring/shared/SwitchSpring.tsx [hunk 1]
```diff
@@ -55,36 +39,27 @@ const SwitchSpring: React.FC<SwitchSpringProps> = ({
...
-            isActive && `bg-lightgray`
-          } h-5`}
-        >
-          <div className={margin}>{icon}</div>
-        </animated.div>
-      </animated.div>
+      {/* div added otherwise tooltip does not work in mantine 6 */}
+      <div>
+        <Switch
+          data-testid="switchSpring-checkbox"
```
Comment (bpierce01) [inline packages/portal-proto/src/components/SwitchSpring/shared/SwitchSpring.tsx]: Can you call this "button-switchspring" instead?

**Review by bpierce01 — CHANGES_REQUESTED** [review]
Change the "switchSpring-checkbox" to "button-switchspring". Thank you!

File: packages/portal-proto/src/components/SwitchSpring/shared/SwitchSpring.tsx (hunk 1 above)
Comment (ocshawn) [reply packages/portal-proto/src/components/SwitchSpring/shared/SwitchSpring.tsx]: yes we can, just FYI the label is applied to the checkbox when the component is rendered not to the button

**Review by bpierce01 — APPROVED** [review]

**Comment by craigrbarnes** [comment]
One issue I noticed is that the original switch has a round control:

image
But this does not:
image

This should be fixed, unless @wteouchicago is ok with it (The documentation  videos have the square buttons)
Additionally, the new control is a bit wider than the original, not sure that is an issue.

**Review by craigrbarnes — CHANGES_REQUESTED** [review]
The switch control is now square, it should be round.

**Comment by wteouchicago** [comment]
Round does seem more ideal (and will match the cancer gene census toggle), but if making it round is a pain, let me know. Also checking @cemile-ctds 's opinion.

Edit: mostly I would prefer the background portion to be the same type of shape as the switch in the foreground.

**Comment by cemile-ctds** [comment]
@craigrbarnes @wteouchicago 
The initial design of the switch buttons on the tables was square. The main reason for this decision is to differentiate these buttons from the on/off buttons on the filters and to customize the column list. Please refer to the table mockup link below and the reference screenshots for more details.

If we are going to follow this design decision, we should also ensure that the background part matches the same shape as the foreground part. 

However, if this is not something we can implement, then please make the switch buttons round; otherwise, please keep the design as it is.

Screenshot 2024-02-22 at 10 51 48


Table mockups: https://xd.adobe.com/view/9494a67c-0198-434e-a925-ac008f31c9b4-c1c7/grid

**Comment by ocshawn** [comment]
@cemile-ctds updated styling to have square background please re evaluate

**Comment by cemile-ctds** [comment]
Screenshot 2024-03-05 at 11 58 10
Look really good @ocshawn. Thank you so much.

**Review by amlehman — APPROVED** [review]

**Review by craigrbarnes — APPROVED** [review]
Looks good!

**Pull request merged**